### PR TITLE
LPD-41375 Replace java8 supported jodd.Paramo dependency with java21 compatible solution

### PR DESCRIPTION
**Pull request by jgarcialr**
see https://liferay.atlassian.net/browse/LPD-41375
//cc @tomibiro

### CODE REVIEW

**Comment by jgarcialr** [comment]
ci:test:relevant

**Comment by jgarcialr** [comment]
ci:test:sf

**Comment by jgarcialr** [comment]
ci:test:security

**Comment by jgarcialr** [comment]
ci:test:sf

**Comment by jgarcialr** [comment]
ci:test:sf

**Comment by jgarcialr** [comment]
ci:test:sf

**Comment by jgarcialr** [comment]
ci:test:relevant

File: modules/apps/portal-remote/portal-remote-json-web-service-web/src/main/java/com/liferay/portal/remote/json/web/service/web/internal/util/ExtractedParameter.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.remote.json.web.service.web.internal.util;
+
+/**
+ * @author Tamas Biro
+ */
+public class ExtractedParameter {
```
Comment (rafaprax) [inline modules/apps/portal-remote/portal-remote-json-web-service-web/src/main/java/com/liferay/portal/remote/json/web/service/web/internal/util/ExtractedParameter.java:11]: Hey @jgarcialr, Can you try to get rid of this class and use the  `MethodParameter` class directly? 

Thx

/cc @tomibiro

**Review by rafaprax — CHANGES_REQUESTED** [review]

File: modules/apps/portal-remote/portal-remote-json-web-service-web/src/main/java/com/liferay/portal/remote/json/web/service/web/internal/util/ExtractedParameter.java (hunk 1 above)
Comment (rafaprax) [reply modules/apps/portal-remote/portal-remote-json-web-service-web/src/main/java/com/liferay/portal/remote/json/web/service/web/internal/util/ExtractedParameter.java:11]: Once you finish this change, I'll review the pr completely. 

Also, please add a unit test for the `MethodParameterExtractor#getMethodParameters`.

Thanks in advance

**Comment by tomibiro** [comment]
Hi @rafaprax and @jgarcialr,
I added the unit tests.
Br, Tamas

**Comment by rafaprax** [comment]
ci:test:sf

**Comment by tomibiro** [comment]
Hi @rafaprax and @jgarcialr,
The source formatter fails, because of some classes seem to be not used in the java test code. I think it is normal, because they are just only for the input Method type in the tests. If you have any solution to make it better please feel free to mention.
Thanks. Tamas

**Comment by jgarcialr** [comment]
ci:test:sf

**Comment by jgarcialr** [comment]
ci:test:relevant

**Comment by jgarcialr** [comment]
ci:test:security

**Comment by jgarcialr** [comment]
ci:test:sf

**Comment by jgarcialr** [comment]
ci:test:relevant

**Comment by jgarcialr** [comment]
ci:test:security

**Comment by jgarcialr** [comment]
ci:test:relevant

**Comment by jgarcialr** [comment]
Waiting on latest result of ci:test:relevant, as previous commit before the force-push had a typo that prevents compilation of some JSONWebServiceTests

**Comment by rafaprax** [comment]
resent here https://github.com/liferay-appsec/liferay-portal/pull/2095

**Comment by rafaprax** [comment]
ci:close

**Pull request closed without merging**